From: Gary Lin <glin@suse.com>
Subject: [PATCH] kabi: ignore struct bpf_verifier_env change
Patch-mainline: No, SUSE-specific
References: bsc#1068032 CVE-2017-5753

patches.fixes/bpf-move-prev_-insn_idx-into-verifier-env.patch moved
'insn_idx' and 'prev_insn_idx' from do_check() to struct bpf_verifier_env
so that those variables can be accessed across several other functions.
To avoid breaking other driver such as nfp, we have to move those two
members to the end of struct.

Signed-off-by: Gary Lin <glin@suse.com>
---
 include/linux/bpf_verifier.h |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

--- a/include/linux/bpf_verifier.h
+++ b/include/linux/bpf_verifier.h
@@ -168,8 +168,6 @@ struct bpf_ext_analyzer_ops {
  * one verifier_env per bpf_check() call
  */
 struct bpf_verifier_env {
-	u32 insn_idx;
-	u32 prev_insn_idx;
 	struct bpf_prog *prog;		/* eBPF program being verified */
 	const struct bpf_verifier_ops *ops;
 	struct bpf_verifier_stack_elem *head; /* stack of verifier states to be processed */
@@ -187,6 +185,10 @@ struct bpf_verifier_env {
 	struct bpf_insn_aux_data *insn_aux_data; /* array of per-insn state */
 
 	struct bpf_verifer_log log;
+#ifndef __GENKSYMS__
+	u32 insn_idx;
+	u32 prev_insn_idx;
+#endif
 };
 
 static inline struct bpf_reg_state *cur_regs(struct bpf_verifier_env *env)
